internal/controller/progressive/progressive.go

```diff
@@ -322,26 +318,31 @@ func processUpgradingChild(
 		return false, false, 0, nil
 
 	case apiv1.AssessmentResultSuccess:
-		// Label the new child as promoted and then remove the label from the old one
-		numaLogger.WithValues("old child", existingPromotedChildDef.GetName(), "new child", existingUpgradingChildDef.GetName(), "replacing 'promoted' child")
-		reasonSuccess := common.LabelValueProgressiveSuccess
-		err := ctlrcommon.UpdateUpgradeState(ctx, c, common.LabelValueUpgradePromoted, &reasonSuccess, existingUpgradingChildDef)
-		if err != nil {
-			return false, false, 0, err
-		}
+		if childStatus.CanDeclareSuccess() {
```

Comment:
I'm looking for `>=AssessUntil` here. Now the call to `CanAssess()` above only looks for whether it's reached the `NextAssessmentTime` (and whether it's previously failed). 

So, basically:
- if we haven't reached `NextAssessmentTime`, we don't assess
- Once we reach `NextAssessmentTime`, we can start assessing. If we see a failure, we can mark it and stop assessing.
- Once we reach `AssessUntil`, we can declare success if we never saw a failure.